### ISSUE TITLE
#115269 broke semver in Fuchsia

### ISSUE THREAD

**Issue by djkoloski** [rust-lang/rust#115852]
We just hit a CI failure in Fuchsia due to #115269 which corrected name resolution ordering for macro-expanded items. This change can break semver for crates. This change probably deserves a crater run to evaluate ecosystem impact, even if the old behavior is incorrect.

**Comment by Urgau** [comment]
Would you happen to have a minimal test case? or something we can reproduce?

Also there was a crater run: https://github.com/rust-lang/rust/pull/115269#issuecomment-1712800245

@rustbot label +T-compiler +E-needs-mcve +regression-untriaged
cc @bvanjoi

**Comment by Nilstrieb** [comment]
I don't quite understand what you mean with "broke semver". Do you mean that it broke Fuchsias build? Or that it changes behavior? Not sure what this has to do with semver, no new releases of crates are being made.

**Comment by djkoloski** [comment]
Sorry for the confusion @Nilstrieb . I say "broke semver" in the sense of changing the API surface in a way that we were able to observe (our build broke). That's mostly based on @obi1kenobi's efforts to track down unexpected semver violations, which has also found a lot of pre-existing problems with glob imports. It's probably not the best terminology for this situation, sorry for the confusion.

We got lucky in the sense that it was just a compiler error instead of a functionality change. I wanted to raise the issue since it seemed unlikely that we were the only ones who observed breakage. The crater run confirms that other crates were affected, but I see now that there was discussion about whether those breakages were acceptable. Since that discussion has already happened, I'm going to close this issue.
